api/envoy/config/tap/v3/common.proto

```diff
@@ -28,7 +29,12 @@ message TapConfig {
 
   // The match configuration. If the configuration matches the data source being tapped, a tap will
   // occur, with the result written to the configured output.
-  MatchPredicate match_config = 1 [(validate.rules).message = {required: true}];
+  // Ignored if :ref:`match <envoy_api_field_config.tap.v3.TapConfig.match>` is set.
+  MatchPredicate match_config = 1 [deprecated = true, (validate.rules).message = {required: true}];
```

Comment:
I followed the https://github.com/envoyproxy/envoy/blob/4f85591457c73a56efb171ec021d1f80b09205fe/api/API_VERSIONING.md#manual-breaking-changes to manually deprecate this field, but to be honest I'm really not sure if this is good, it looks weird since it's a required field. 

I also tried to change the type directly to `common.matcher.v3.MatchPredicate`, this doesn't break the wire compatibility and is cleaner in the API, but I think it breaks the control plane code because the type is changed.

@mattklein123 @lizan please let me know if you have any suggestions for this, thank you.

Reply:
> I also tried to change the type directly to common.matcher.v3.MatchPredicate, this doesn't break the wire compatibility and is cleaner in the API, but I think it breaks the control plane code because the type is changed.

Right, unfortunately this is no longer allowed.

> but to be honest I'm really not sure if this is good, it looks weird since it's a required field.

Yeah this won't work. You will need to remove the required validation which should be fine since it's strictly looser, and then have error handling in the code.

This will also need a release note.

Reply:
Thanks, updated release note.